app/main.py

```diff
@@ -1,2 +1,56 @@
+from collections.abc import Hashable
+from typing import Any
+
+
 class Dictionary:
-    pass
+    def __init__(
+            self,
+            initial_capacity: int = 8,
+            load_factor: float = 0.75
+    ) -> None:
+        self.load_factor = load_factor
+        self.capacity = initial_capacity
+        self.size = 0
+        self.table = [[] for _ in range(self.capacity)]
+
+    def _hash(self, key: Hashable) -> int:
+        return hash(key) % self.capacity
+
+    def _resize(self) -> None:
+        old_table = self.table
+        self.capacity *= 2
+        self.table = [[] for _ in range(self.capacity)]
+        self.size = 0
+
+        for entry in old_table:
+            if entry:
+                for k, v, h in entry:
+                    self.__setitem__(k, v)
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        if self.size >= self.capacity * self.load_factor:
+            self._resize()
+
+        index = self._hash(key)
+
+        for i, (k, v, h) in enumerate(self.table[index]):
+            if k == key:
+                self.table[index][i] = (key, value, h)
+                return
```

Comment:
Overall logic look correct you just need to refactor code so you use `Nodes` instead of lists to better structure your code